### PR TITLE
Symmetric tree

### PR DESCRIPTION
**Pull request by JeevaRamanathan**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: data_structures/binary_tree/symmetric_tree.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+class Node:
+    """
+    A Node has data variable and pointers to Nodes to its left and right.
+    """
+
+    def __init__(self, data: int) -> None:
+        self.data = data
+        self.left: Node | None = None
+        self.right: Node | None = None
```
Comment (cclauss) [inline data_structures/binary_tree/symmetric_tree.py:17]:
```suggestion
from dataclasses import dataclass


@dataclass
class Node:
    """
    A Node has a data variable and pointers to Nodes to its left and right.
    """
    data: int
    left: Node | None = None
    right: Node | None = None
```

File: data_structures/binary_tree/symmetric_tree.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+           / \    / \
+          3   4   4  3
+
+        >>> tree = Node(1)
+        >>> tree.left = Node(2)
+        >>> tree.right = Node(2)
+        >>> tree.left.left = Node(3)
+        >>> tree.left.right = Node(4)
+        >>> tree.right.left = Node(4)
+        >>> tree.right.right = Node(3)
```
Comment (cclauss) [inline data_structures/binary_tree/symmetric_tree.py:38]: Please move these lines into a function defined as `def make_symmetric_tree() -> Node:`.

File: data_structures/binary_tree/symmetric_tree.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+         / \    / \
+        3   4   3  4
+
+        >>> tree2 = Node(1)
+        >>> tree2.left = Node(2)
+        >>> tree2.right = Node(2)
+        >>> tree2.left.left = Node(3)
+        >>> tree2.left.right = Node(4)
+        >>> tree2.right.left = Node(3)
+        >>> tree2.right.right = Node(4)
```
Comment (cclauss) [inline data_structures/binary_tree/symmetric_tree.py:56]: Please move these lines into a function defined as `def make_asymmetric_tree() -> Node:`.

File: data_structures/binary_tree/symmetric_tree.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+    def is_mirror(self, left: Node | None, right: Node | None) -> bool:
+        """
+        >>> tree1 = Node(1)
+        >>> tree1.left = Node(2)
+        >>> tree1.right = Node(2)
+        >>> tree1.left.left = Node(3)
+        >>> tree1.left.right = Node(4)
+        >>> tree1.right.left = Node(4)
+        >>> tree1.right.right = Node(3)
```
Comment (cclauss) [inline data_structures/binary_tree/symmetric_tree.py:74]: Please move these lines into a function defined as `def make_mirror_tree() -> Node:`.

**Review by cclauss — CHANGES_REQUESTED** [review]

File: data_structures/binary_tree/symmetric_tree.py (hunk 4 above)
Comment (cclauss) [inline data_structures/binary_tree/symmetric_tree.py:74]:
```suggestion
        >>> tree1 = make_symmetric_tree()
```

File: data_structures/binary_tree/symmetric_tree.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        >>> SymmetricTree().is_mirror(tree1.left, tree1.right)
+        True
+        """
```
Comment (cclauss) [inline data_structures/binary_tree/symmetric_tree.py:77]:
```suggestion
        >>> tree2 = make_asymmetric_tree()
        >>> SymmetricTree().is_mirror(tree2.left, tree2.right)
        False
        """
```

File: data_structures/binary_tree/symmetric_tree.py [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+        >>> tree = Node(1)
+        >>> tree.left = Node(2)
+        >>> tree.right = Node(2)
+        >>> tree.left.left = Node(3)
+        >>> tree.left.right = Node(4)
+        >>> tree.right.left = Node(4)
+        >>> tree.right.right = Node(3)
+
+        >>> SymmetricTree().is_symmetric_tree(tree)
+        True
```
Comment (cclauss) [inline data_structures/binary_tree/symmetric_tree.py:41]:
```suggestion
        >>> SymmetricTree().is_symmetric_tree(make_symmetric_tree())
        True
        >>> SymmetricTree().is_symmetric_tree(make_asymmetric_tree())
        False
```

File: data_structures/binary_tree/symmetric_tree.py (hunk 2 above)
Comment (JeevaRamanathan) [reply data_structures/binary_tree/symmetric_tree.py:38]: there is no function named as  make_symmetric_tree(), are you expecting to create a new function in the name of  make_symmetric_tree(), can you please clarify

Comment (cclauss) [reply data_structures/binary_tree/symmetric_tree.py:38]: Yes.  Please create the function.

Comment (cclauss) [reply data_structures/binary_tree/symmetric_tree.py:38]: The reason that we want separate functions is that building the trees takes a lot of space and complexity and it is not really where we want visitors to focus their attention.  Also, by creating separate tree-building functions we can reuse them in multiple tests.

**Comment by JeevaRamanathan** [comment]
Hi cclauss,
I have made the requested changes. Please check and approve.

regards,
Jeeva Ramanathan M

File: data_structures/binary_tree/symmetric_tree.py (hunk 6 above)
Comment (JeevaRamanathan) [reply data_structures/binary_tree/symmetric_tree.py]: @cclauss  I have made the requested changes, please check

**Comment by cclauss** [comment]
I think this should pass...

**Review by cclauss — APPROVED** [review]
Looks great.  I dropped the class because we were not really using it.

**Pull request merged**